lib/models/update-checker.js

```diff
@@ -56,8 +56,16 @@ UpdateChecker.prototype.doCheck = function() {
   }.bind(this)).catch(function() {
     return this.checkNPM();
   }.bind(this)).then(function(version) {
+    // match postfix SHA in dev version
+    var isDevBuild   = !!this.localVersion.match(/\b[0-9a-f]{5,40}\b/);
```

Comment:
I think I'd rather make [lib/utilities/ember-cli-version.js](https://github.com/ember-cli/ember-cli/blob/master/lib/utilities/ember-cli-version.js) be the only thing that has to be aware of this.  Can we make that script export another function (like `isDevelopment()`) then just consume it here?  

I'm mostly concerned with having the concept of what exactly goes into a dev build version number  be encapsulated in a single place.